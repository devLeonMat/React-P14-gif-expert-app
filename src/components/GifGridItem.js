import React from "react";
import 'animate.css';
import PropTypes from "prop-types";

export const GifGridItem = ({title, url}) => {
    return (<div className="card animate__fadeIn">
        <img src={url} alt={title}/>
        <p>{title}</p>
    </div>);
}


GifGridItem.prototypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
