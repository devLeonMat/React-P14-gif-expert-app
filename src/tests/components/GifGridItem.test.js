import React from "react";
import {shallow} from "enzyme";
import {GifGridItem} from "../../components/GifGridItem";

describe('test in <GifGridItem/>', () => {

    const title = 'title';
    const url = 'https://localhost/something.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('should display the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('must have a title paragraph', () => {
        expect(wrapper.find('p').text().trim()).toBe(title);
    });

    test('must have an image', () => {
        expect(wrapper.find('img').prop('src')).toBe(url);
        expect(wrapper.find('img').prop('alt')).toBe(title);
    });

    test('must have animate__fadeIn', () => {
        expect(wrapper.find('div').prop('className')).toContain('animate__fadeIn');
    });


})
