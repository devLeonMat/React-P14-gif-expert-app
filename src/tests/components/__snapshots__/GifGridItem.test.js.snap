// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifGridItem/> should display the component correctly 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="title"
    src="https://localhost/something.jpg"
  />
  <p>
    title
  </p>
</div>
`;
